### DISCUSSION TITLE
v11 Beta 3

### DISCUSSION THREAD

**Discussion by joshblack** [carbon-design-system/carbon#10247] [Release Feedback]
Have you tried out the v11 Beta 23 release and have any feedback? We'd love to hear it! From issues that you might have run into to questions you might have, feel free to post anything release-related here.

Thanks so much for your help! We really appreciate it 🙏

**Comment by mattrosno** [comment]
Using `@carbon/react` 0.11.0. Search component:

```jsx
        <Search
          id="catalog-search"
          labelText="Search component index by name, keyword, or domain"
          placeholder="Component name, keyword, domain"
          value={search}
          onBlur={handleOnBlur}
          onChange={handleOnChange}
          onClear={handleOnClear}
          size="lg"
        />
```

That's rendering at 40px tall not 48px, even though the `--lg` class name is getting applied. Per the migration guide, `lg` should be 48px. https://v11.carbondesignsystem.com/migrating/guide/develop#size-prop-updates-breaking

Storybook doesn't have any knobs so I can't verify the bug. https://carbon-react-next.netlify.app/?path=/story/components-search--default

![image](https://user-images.githubusercontent.com/1691245/145636033-56d56365-2021-43a9-8098-01e01620c5d6.png)

**Reply by joshblack** [reply]
Issue: https://github.com/carbon-design-system/carbon/issues/10295

**Comment by mattrosno** [comment]
Pagination next/previous missing icons in 0.11.0. https://carbon-react-next.netlify.app/?path=/story/components-pagination--pagination

![image](https://user-images.githubusercontent.com/1691245/145637201-95bf0fbc-dc63-451a-b0dd-737bfcfe1b56.png)

**Reply by joshblack** [reply]
Issue: https://github.com/carbon-design-system/carbon/issues/10294

**Comment by mattrosno** [comment]
UI shell header action icons aren't centered in 0.11.0. https://carbon-react-next.netlify.app/?path=/story/components-ui-shell--header-base-w-actions-and-right-panel

![image](https://user-images.githubusercontent.com/1691245/145637572-e888ca1a-3027-4c12-8cd6-626d5b085fb7.png)

**Reply by joshblack** [reply]
Issue: https://github.com/carbon-design-system/carbon/issues/10294

**Comment by mattrosno** [comment]
The new `IconButton` does not pass the `size` prop to the button, and instead appears to hardcode the `sm` button. To render at 48px, I had to override.

```jsx
            <IconButton
              className={clsx(styles.button, view === 'grid' && styles.selected)}
              kind="ghost"
              label="Grid view"
              onClick={() => {
                onView('grid')
              }}
              size="lg"
            >
              <GridIcon size={20} />
            </IconButton>
```

```scss
.button {
  // override carbon because `lg` isn't supported
  width: $spacing-09;
  height: $spacing-09;

  // override icon alignment because the IconButton only supports `sm` size currently
  &:global(.cds--btn--sm) {
    padding-right: 14px;
    padding-left: 14px;
  }
}
```

**Reply by joshblack** [reply]
Issue for reference: https://github.com/carbon-design-system/carbon/issues/10293

**Comment by Eugeno** [comment]
#### Questions and suggestions about tokens

**text-error** is `red-30` on `gray-100` looks _too pink_, how about making it `red-40`? It would be consistent with `support-error` which has different colors for `gray-90` and `gray-100`.

**highlight** looks strange: `blue-20` for `white`, `gray-10`, `gray-100` and `blue-70` for `gray-90`.

**toggle-off**: `gray-50` on `gray-100` so shiny, I suggest `gray-60`.

**border-disabled**: at every theme it is more visible and clear than `border-subtle` (problems for accordion with disabled item).

**Reply by joshblack** [reply]
cc @aagonzales

**Reply by aagonzales** [reply]
**text-error** has to pass 4.5:1 text contrast across several background colors which is why its lighter then `support-error` which only needs to pass non-text contrast at 3:1. 

**Highlight**: oops you found a bug there for the gray100 value. That should be `Blue80` in the Gray 100 theme. Its supposed to be subtle, see the date range in date picker for its use.

**Toggle-off**: Another bug, its actually is supposed to be `Gray-60` in the Gray100 theme.

**Border-disabled:** The rule (which we don't really have documented) is that border-subtle actually doesn't change colors on disabled. Border-disabled is actually more for items that use `$border-interactive` or $border-strong`

**Reply by aagonzales** [reply]
@joshblack seems like we need to include another color token QA audit for the RC release.
Link to table with most accurate intended values: https://carbondesignsystem.notion.site/final-List-of-new-tokens-c507394e43fe4daa9f0cd0eedb25d1e4

**Reply by aagonzales** [reply]
Actually just took a look at `text-error` again and I think we can actually do `Red 40` in the Gray 100 theme. 
![image](https://user-images.githubusercontent.com/11670886/146076103-0fe3b479-6330-4c73-b002-ec5f84be4b58.png)

**Reply by Eugeno** [reply]
Thanks for your feedback!

Looking at your table in notion, I would like to highlight two more potential problems: `layer-accent-active-02` and `layer-active-03`, both at `Gray 100`.

**Reply by aagonzales** [reply]
Yeah those seem weird because they're following the [rules for active states](https://v11.carbondesignsystem.com/guidelines/color/overview/#active) and the at the 60 step in the scale colors start to darken instead if lighten. I noted it as weird when I made those value and I've been waiting to see how it plays out when people use it. It is pretty nested in the layering model at that point so I'm not sure how much it gets used. But I wanted to see some real examples before we bent the color rules.

**Reply by joshblack** [reply]
Issue for reference: https://github.com/carbon-design-system/carbon/issues/10292

**Comment by ricardo-valero** [comment]
This is irrelevant from the beta, but I thought it might be easier to consider before an official release

I mentioned this in the svelte components repo a while ago
https://github.com/carbon-design-system/carbon-components-svelte/issues/709

But I'd like to know what does the main library thinks about naming components and conventions, for example: 

* FormGroup *(rename to InputGroup? it can be misinterpreted as a group of forms, which is obviously not... see Bootstrap's naming)*
* FormItem *(rename to InputItem?)*
* FormLabel *(rename to Label?)*
* RadioTile and SelectableTile are RadioButton and Checkbox equivalents
  * Proposal: RadioButtonTile, CheckboxTile
* Use only one word to represent groups (Set, Group or plural form)
  * Current: ButtonSet, RadioButtonGroup, Tabs
  * Proposal: ButtonGroup, RadioButtonGroup, TabGroup
  
I get that it isn't very important for the end user, I just wish to contribute in some way.

**Reply by Eugeno** [reply]
_A FormGroup aggregates the values of each child FormControl into one object_ (see Angular's naming) 😄

**Reply by ricardo-valero** [reply]
But with that definition shouldn't it be *FormControlGroup*? 
I'd say that's inconsistent.

Why is 
*RadioButtonGroup* a group of *RadioButtons* 
but 
*FormGroup* **not** a group of *Forms*?

That is basically what I was trying to say.

**Comment by Eugeno** [comment]
![image](https://user-images.githubusercontent.com/23382920/147193048-c8950128-9687-4275-a060-10da94b86707.png)
https://v11.carbondesignsystem.com/components/text-input/usage/ password input

**Comment by ricardo-valero** [comment]
Side nav link with icon looks great

<img width="257" alt="image" src="https://user-images.githubusercontent.com/55701657/147499686-e0c177af-e586-4e50-863a-fdf5f39739af.png">

```jsx
<SideNavLink href="/" renderIcon={Home}>
    Home
</SideNavLink>
```

Unfortunately button with icon doesn't

<img width="135" alt="image" src="https://user-images.githubusercontent.com/55701657/147500214-c9a128f2-6ff5-4283-85a8-851231467336.png">

```jsx
<Button renderIcon={Home}>
    Home
</Button>
```

**Reply by ricardo-valero** [reply]
Works with `@carbon/react 0.13.0-rc.0`

**Comment by Eugeno** [comment]
Tokens for buttons also need careful validation.
Example: primary danger button (https://v11.carbondesignsystem.com/components/button/style/)
<html>
<body>


Container | background-color | $button-danger-primary
-- | -- | --
Container:hover | background-color | $button-danger-hover


</body>
</html>

gray90 (https://carbondesignsystem.notion.site/final-List-of-new-tokens-c507394e43fe4daa9f0cd0eedb25d1e4):
<html>
<body>


button-danger-primary | Red 60
-- | -- 
button-danger-hover | Red-40-hover


</body>
</html>

Base `red-60` and `red-40-hover` for hover, doubtful.

Btw, how about change `focus-inset` at `gray10` and `gray90` into `gray-10` and `gray-90` (instead of `white` and `gray-100`)? Imo looks better for tertiary.

**Reply by joshblack** [reply]
cc @aagonzales

**Reply by aagonzales** [reply]
Ah yes good catch that hover should be `Red-60-Hover`. The v11 button colors have been hard to validate and reference as they are not part of this [element preview](https://carbon-elements.netlify.app/themes/examples/preview-v11/). @joshblack can we add the button tokens to the element preview or is there another place to easily inspect the button token colors?

**Reply by aagonzales** [reply]
It looks like the [v11 danger button](https://carbon-react-next.netlify.app/?path=/story/components-button--danger&globals=theme:g90) is rendering correctly so it was probably just an error in the Notion table but looks like its correct in the code. I'll try and track down the button token colors in code and confirm.

**Reply by Eugeno** [reply]
I see difference in focus state for ghost danger button in [v11 danger button](https://carbon-react-next.netlify.app/?path=/story/components-button--danger&globals=theme:g90) and https://v11.carbondesignsystem.com/components/button/style/#ghost-danger-button

**Reply by Eugeno** [reply]
Looks like problems in the first link, And `active` doesn't work for tertiary.

**Reply by Eugeno** [reply]
And more: tertiary, danger tertiary and danger ghost at disabled state have `text-on-color-disabled` while ghost has `text-disabled`.
Tertiary has a wrong description about label color at hover (active, focus) — `text-on-color`, should be `text-inverse`, I suppose.

**Comment by mattrosno** [comment]
I'm using `@include type-style('display-01', true);` with `"@carbon/react": "^0.12.0"` and getting a deprecation warning.

![image](https://user-images.githubusercontent.com/1691245/151875334-dbd4c490-d66b-4110-aec3-34a5fac2e82e.png)

Which is coming from this line in the type package.

https://github.com/carbon-design-system/carbon/blob/c7e1cdf6d5f869b971b10ebbb5aa1c8f3b49cdc1/packages/type/scss/modules/_styles.scss#L699

**Reply by joshblack** [reply]
@mattrosno PR up: https://github.com/carbon-design-system/carbon/pull/10591

**Reply by mattrosno** [reply]
Thanks! 🚀
